dbt/adapters/bigquery/retry.py

```diff
@@ -74,6 +73,11 @@ def __init__(self, retries: int) -> None:
         self._retries: int = retries
         self._error_count = 0
 
+    @staticmethod
+    def _is_retryable(error: Exception) -> bool:
```

Comment:
If we decide to remove `BadRequest`, this should go away and we should call `_job_should_retry` directly. We may decide to remove `BadRequest` since we would likely expect all retries to fail as well given the error appears to be on the user's side (malformed request) rather than BQ's side (e.g. resource unavailable).